akka-persistence/src/main/scala/akka/persistence/Persistence.scala

```diff
@@ -146,6 +147,16 @@ class Persistence(val system: ExtendedActorSystem) extends Extension {
 
   private val config = system.settings.config.getConfig("akka.persistence")
 
+  /**
+   * INTERNAL API: When starting many persistent actors at the same time the journal
+   * its data store is protected from being overloaded by limiting number
+   * of recoveries that can be in progress at the same time.
+   */
+  @InternalApi private[akka] val recoveryPermitter: ActorRef = {
+    val maxPermits = config.getInt("max-concurrent-recoveries")
+    system.systemActorOf(RecoveryPermitter.props(maxPermits), "recoveryPermitter")
+  }
```

Comment:
Just a thought: no need to fix it, but sharing the thought:
So if someone configured multiple plugins, the replay at the same time is shared between all journals.

Since this is internal API we can improve this if needed, so ok to keep the simple shared limit now I think